### PR TITLE
Bug 1940207: create the ovs-config-executed file to signal ovs is running on the host

### PR DESCRIPTION
**Pull request by fedepaol**
**- What I did**
The file is not needed by 4.7+ ovs pods, but when rolling back from 4.7
to 4.6 CNO is updated before MCO. The 4.6 version of the pod does not
find the file, executes ovs in a pod with the result that two ovs
instances are running at the same time.

**- How to verify it**

Need to test the downgrade with this PR in

**- Description for the changelog**

### CODE REVIEW

**Comment by fedepaol** [comment]
/hold
Need to verify it really fixes the issue

**Comment by fedepaol** [comment]
I managed to test this.
Spawned a 4.7 cluster, overrode the mco following the instructions in https://github.com/openshift/machine-config-operator/blob/master/docs/HACKING.md#running-in-a-cluster , forced the downgrade to 4.6

`oc adm upgrade --to-image=registry.ci.openshift.org/ocp/release:4.6.0-0.ci-2021-03-31-003732 --allow-explicit-upgrade --force`

ovs says it's running in systemd:

```
oc -n openshift-sdn logs ovs-cdd9b
openvswitch is running in systemd
==> /host/var/log/openvswitch/ovs-vswitchd.log <==
2021-04-06T14:01:45.335Z|00403|connmgr|INFO|br0<->unix#1003: 2 flow_mods in the last 0 s (2 deletes)
2021-04-06T14:01:45.377Z|00404|connmgr|INFO|br0<->unix#1006: 4 flow_mods in the last 0 s (4 deletes)
2021-04-06T14:01:45.416Z|00405|bridge|INFO|bridge br0: deleted interface vethb03eb98a on port 10
2021-04-06T14:01:55.710Z|00406|bridge|INFO|bridge br0: added interface veth1cc74392 on port 57
```

Upgrade is progressing:
```
➜  Downloads oc get clusteroperators           
NAME                                       VERSION                        AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication                             4.6.0-0.ci-2021-04-02-023150   True        False         False      23m
baremetal                                  4.7.0-0.ci-2021-04-02-034219   True        False         False      171m
cloud-credential                           4.6.0-0.ci-2021-04-02-023150   True        False         False      177m
cluster-autoscaler                         4.6.0-0.ci-2021-04-02-023150   True        False         False      170m
config-operator                            4.6.0-0.ci-2021-04-02-023150   True        False         False      171m
console                                    4.6.0-0.ci-2021-04-02-023150   True        False         False      22m
csi-snapshot-controller                    4.6.0-0.ci-2021-04-02-023150   True        False         False      69m
dns                                        4.6.0-0.ci-2021-04-02-023150   True        False         False      161m
etcd                                       4.6.0-0.ci-2021-04-02-023150   True        False         False      170m
image-registry                             4.6.0-0.ci-2021-04-02-023150   True        False         False      163m
ingress                                    4.6.0-0.ci-2021-04-02-023150   True        False         False      162m
insights                                   4.6.0-0.ci-2021-04-02-023150   True        False         False      164m
kube-apiserver                             4.6.0-0.ci-2021-04-02-023150   True        False         False      169m
kube-controller-manager                    4.6.0-0.ci-2021-04-02-023150   True        False         False      169m
kube-scheduler                             4.6.0-0.ci-2021-04-02-023150   True        False         False      169m
kube-storage-version-migrator              4.6.0-0.ci-2021-04-02-023150   True        False         False      73m
machine-api                                4.6.0-0.ci-2021-04-02-023150   True        False         False      164m
machine-approver                           4.6.0-0.ci-2021-04-02-023150   True        False         False      170m
machine-config                             4.7.0-0.ci-2021-04-02-034219   False       True          False      6m51s
marketplace                                4.6.0-0.ci-2021-04-02-023150   True        False         False      22m
monitoring                                 4.6.0-0.ci-2021-04-02-023150   True        False         False      21m
network                                    4.6.0-0.ci-2021-04-02-023150   True        False         False      171m
node-tuning                                4.6.0-0.ci-2021-04-02-023150   True        False         False      22m
openshift-apiserver                        4.6.0-0.ci-2021-04-02-023150   True        False         False      23m
openshift-controller-manager               4.6.0-0.ci-2021-04-02-023150   True        False         False      98m
openshift-samples                          4.6.0-0.ci-2021-04-02-023150   True        False         False      22m
operator-lifecycle-manager                 4.6.0-0.ci-2021-04-02-023150   True        False         False      170m
operator-lifecycle-manager-catalog         4.6.0-0.ci-2021-04-02-023150   True        False         False      170m
operator-lifecycle-manager-packageserver   4.6.0-0.ci-2021-04-02-023150   True        False         False      22m
service-ca                                 4.6.0-0.ci-2021-04-02-023150   True        False         False      171m
storage                                    4.6.0-0.ci-2021-04-02-023150   True        False         False      69m
```
The upgrade is stuck on MCO with the following error:
```
  extension:
    lastSyncError: 'pool master has not progressed to latest configuration: controller version mismatch for rendered-master-cb2db7df54e993c796b76a2242b3e08a expected d5dc2b519aed5b3ed6a6ab9e7f70f33740f9f8af has b5723620cfe40e2e4e8cbdcb105d6ae534be1753: pool is degraded because rendering fails with "": "Failed to render configuration for pool master: parsing Ignition config failed: unknown version. Supported spec versions: 2.2, 3.0, 3.1", retrying'
    master: 'pool is degraded because rendering fails with "": "Failed to render configuration for pool master: parsing Ignition config failed: unknown version. Supported spec versions: 2.2, 3.0, 3.1"'
    worker: 'pool is degraded because rendering fails with "": "Failed to render configuration for pool worker: parsing Ignition config failed: unknown version. Supported spec versions: 2.2, 3.0, 3.1"'
```

But I think this fixed the network part.

**Comment by fedepaol** [comment]
test e2e-aws-serial

**Comment by trozet** [comment]
@kikisdeliveryservice @zhaozhanqi any feedback on the MCO error @fedepaol saw here?
https://github.com/openshift/machine-config-operator/pull/2506#issuecomment-814168869

**Review by trozet — APPROVED** [review]
/lgtm

**Comment by yuqi-zhang** [comment]
The error is expected. The MCO doesn't support downgrades, so the 4.6 MCO doesn't understand how to parse ignition 3.2 configs (4.7)

**Comment by fedepaol** [comment]
> The error is expected. The MCO doesn't support downgrades, so the 4.6 MCO doesn't understand how to parse ignition 3.2 configs (4.7). This in turn means unfortunately all downgrades from 4.7->4.6 will fail

Ugh. I found this while investigating the failure of a downgrade job. I guess that does not make sense then?
See https://bugzilla.redhat.com/show_bug.cgi?id=1940207

**Comment by yuqi-zhang** [comment]
Commented in BZ. I am under the impression downgrades are not supported

**Comment by yuqi-zhang** [comment]
/approve

The fix itself lgtm if it fixes the networking problem

**Pull request merged**
